rotkehlchen/db/calendar.py

```diff
@@ -234,3 +276,51 @@ def update_calendar_entry(self, calendar: CalendarEntry) -> int:
                 raise InputError(f'Could not update calendar entry due to {e}') from e
 
         return calendar.identifier
+
+    def create_reminder_entry(self, entry: ReminderEntry) -> int:
+        """Insert the provided event reminder in the database.
+        May raise:
+        - InputError if any db constraint is not satisfied
+        """
+        with self.db.user_write() as write_cursor:
+            try:
+                write_cursor.execute(
+                    'INSERT OR IGNORE INTO calendar_reminders(event_id, secs_before) '
+                    'VALUES (?, ?) RETURNING identifier',
+                    entry.serialize_for_db()[:-1],  # exclude the default identifier since we need to create it  # noqa: E501
+                )
+            except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
+                raise InputError(f'Could not add reminder entry due to {e}') from e
+
+            if (identifier_row := write_cursor.fetchone()) is None:
+                raise InputError(
+                    'Could not add the reminder entry because an event with the same name, '
+                    'address and blockchain already exist',
+                )
+
+            return identifier_row[0]
+
+    def update_reminder_entry(self, reminder: ReminderEntry) -> int:
+        """Update reminder in the database
+        - InputError if no event reminder got updated
+        """
+        with self.db.user_write() as write_cursor:
+            try:
+                write_cursor.execute(
+                    'UPDATE calendar_reminders SET event_id=?, secs_before=? WHERE identifier=?',
+                    reminder.serialize_for_db(),
+                )
+            except sqlcipher.IntegrityError as e:  # pylint: disable=no-member
+                raise InputError(f'Could not update reminder due to {e}') from e
+
+        return reminder.identifier
+
+    def query_reminder_entry(self, event_id: int) -> dict[str, list[ReminderEntry]]:
+        """Query reminder using the id of the event linked"""
```

Comment:
Also explain in the docstring what the order they are returned in is.

```suggestion
        """Query reminder using the id of the linked event"""
```